receiver/hostmetricsreceiver/internal/scraper/processscraper/config.go

```diff
@@ -31,10 +31,10 @@ type Config struct {
 	Include MatchConfig `mapstructure:"include"`
 	Exclude MatchConfig `mapstructure:"exclude"`
 
-	// MuteProcessNameError is a flag that will mute the error encountered when trying to read a process the
-	// collector does not have permission for.
+	// MuteProcessErrors is a flag that will mute the error encountered when trying to read a process the
+	// collector does not have permission for or try to read parent process on k8s environments which does not exist.
 	// See https://github.com/open-telemetry/opentelemetry-collector/issues/3004 for more information.
-	MuteProcessNameError bool `mapstructure:"mute_process_name_error,omitempty"`
+	MuteProcessErrors bool `mapstructure:"mute_process_errors,omitempty"`
```

Comment:
Just replacing a config option is a significant breaking change. We need to support both for a few versions with `MuteProcessNameError` marked as deprecated and throwing a warning when it's set to true in user settings.

Reply:
refactored as suggested